### PR TITLE
[AIRFLOW-4928] Move all config parses to class properties

### PR DESCRIPTION
**Pull request by serkef**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-4928

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Every `airflow.configuration.conf` method is being called multiple times. We can avoid redundant evaluation by storing the parsed value in the class namespace.

### Tests

- [x] My PR does not need testing for this extremely good reason:
Is already covered. No new features are implemented.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/models/dagbag.py [hunk 1]
```diff
@@ -72,13 +73,18 @@ class DagBag(BaseDagBag, LoggingMixin):
...
+    dagbag_import_timeout = conf.getint('core', "DAGBAG_IMPORT_TIMEOUT")
+    unit_test_mode = conf.getboolean('core', 'unit_test_mode')
+    scheduler_zombie_task_threshold = conf.getint('scheduler', 'scheduler_zombie_task_threshold')
 
     def __init__(
             self,
             dag_folder=None,
             executor=None,
-            include_examples=configuration.conf.getboolean('core', 'LOAD_EXAMPLES'),
-            safe_mode=configuration.conf.getboolean('core', 'DAG_DISCOVERY_SAFE_MODE')):
```
Comment (ashb) [inline airflow/models/dagbag.py]: Default arguments for python functions are evaluated at parse time, not run time, so this hunk isn't needed

**Comment by ashb** [comment]
Did you notice a performance improvment from doing this, or is it just "this way is better"?

**Comment by serkef** [comment]
@ashb I don't have a benchmark. The default arguments are evaluated in parsing, but still we don't need to evaluate multiple times. Also, two changes are in functions' bodies which is executed multiple times. (One is in a loop for every task). Configuration doesn't store the values (in that case it would be "same" cost to retrieve it from there), but instead it is only storing the plaintext and it's evaluating and "parsing" each variable on each call - I'm not 100% sure but that's what I got from skimming through the code. 
Let me know if you think this doesn't provide any value and will close it.

File: airflow/models/dagbag.py [hunk 2]
```diff
@@ -194,7 +200,7 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
             if mod_name in sys.modules:
                 del sys.modules[mod_name]
 
-            with timeout(configuration.conf.getint('core', "DAGBAG_IMPORT_TIMEOUT")):
+            with timeout(self.dagbag_import_timeout):
```
Comment (ashb) [inline airflow/models/dagbag.py]: Let's make these class-level vars be constants - `self.DAGBAG_IMPORT_TIMEOUT` etc.

File: airflow/models/dagbag.py (hunk 1 above)
Comment (ashb) [reply airflow/models/dagbag.py]: And let's revert this

File: airflow/models/dagbag.py (hunk 2 above)
Comment (serkef) [reply airflow/models/dagbag.py]: good point

**Comment by serkef** [comment]
@ashb Fixed your comments and rebased. Please take another look.

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
Just made a few cosmetic changes. Will merge once it's green. (Feel free to ping me if I forget again!)

**Comment by serkef** [comment]
@ashb It's good to go

**Pull request merged**